### DISCUSSION TITLE
Having too many windows on workspace makes it slow to load and buggy

### DISCUSSION THREAD

**Discussion by SolsticeSpectrum** [i3/i3#5890] [Q&A]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
When you have too many windows open on one workspace, opening it takes like two seconds and the rendering can get really messy for a second
The number of windows I tested it with is **64**

## Expected Behavior
It should be optimized more

## Reproduction Instructions
Open hell a lot of windows on one workspace and then switch to different one and back

## Environment
Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.23-7-g69f68dcd+ © 2009 Michael Stapelberg and contributors
Running i3 version: 4.23-7-g69f68dcd+ (pid 16325)
Loaded i3 config:
  /home/ruda0/.config/i3/config (main) (last modified: Po 15. ledna 2024, 21:46:01, 696901 seconds ago)

The i3 binary you just called: /usr/local/bin/i3
The i3 binary you are running: i3
</pre>

<details><summary>Config file</summary><pre>
https://termbin.com/on8a
</pre>
</details>

<pre>
- Linux Distribution & Version: Arch Linux latest rolling
- Are you using a compositor (e.g., xcompmgr or compton): picom
</pre>

![image](https://github.com/i3/i3/assets/38611351/836cc173-4966-4c4b-981d-a4d2ea55c2a6)
![image](https://github.com/i3/i3/assets/38611351/84f205ad-e862-401f-8ade-1158395b0cf5)

**Comment by okraits** [comment]
I can't reproduce it. Works fine here.
Are you running a compositor?
Which terminal emulator do you use? Does it happen with something like xterm as well?
Did you try that with a different window manager like openbox?
Could also be an issue with your graphics driver.

**Comment by orestisfl** [comment]
Try reproducing without picom

**Comment by SolsticeSpectrum** [comment]
> Try reproducing without picom
Same result with picom disabled, maybe Nvidia issue? I use kitty terminal.
